File: CSharpServer/Qipai4Core/Qipai4Core/net/silverfoxserver/core/socket/SessionMessage.cs
/*
* SilverFoxServer: massive multiplayer game server for Flash, ...
* VERSION:3.0
* PUBLISH DATE:2015-9-2 
* GITHUB:github.com/wdmir/521266750_qq_com
* UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
* COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
* MAIL:521266750@qq.com
*/
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;
using SuperSocket.SocketBase;

namespace net.silverfoxserver.core.socket
{
    public class SessionMessage
    {

        private AppSession _e = null;

        private String _strIpPort = string.Empty;

        private XmlDocument _doc = null;
        
        private bool _fromRCServer = false;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="strIpPort_">ip</param>
        /// <param name="doc_"></param>
        /// <param name="fromServer_">是否是同组服务器发来的消息，因现在是共用一个处理队列，防止客户端伪造</param>
        public SessionMessage(AppSession e_, XmlDocument doc_,
            bool fromRCServer_,String strIpPort_)
        {

            _e = e_;

            _strIpPort = strIpPort_;

            _doc = doc_;


            _fromRCServer = fromRCServer_;
        }

        public AppSession e()
        {
            return _e;
        }
        
        public XmlDocument doc()
        {
            return _doc;
        }

        /// <summary>
        /// 是否是同组服务器发来的消息，因现在是共用一个处理队列，防止客户端伪造
        /// </summary>
        /// <returns></returns>
        public bool fromServer
        {
           get {

                return _fromRCServer;             

           }
            
        }

        public string action()
        {
            return _doc.DocumentElement.ChildNodes[0].Attributes["action"].Value;
        }


        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public string strIpPort()
        {
            if (_e != null) { 
                return _e.getRemoteEndPoint().ToString();
            }

            return _strIpPort;
        }

        public void Dereference()
        {

            _e = null;
            _doc = null;

        }

    }



}

File: JavaServer/Qipai4Core/src/net/silverfoxserver/core/socket/SessionMessage.java
/*
 * SilverFoxServer: massive multiplayer game server for Flash, ...
 * VERSION:3.0
 * PUBLISH DATE:2015-9-2 
 * GITHUB:github.com/wdmir/521266750_qq_com
 * UPDATES AND DOCUMENTATION AT: http://www.silverfoxserver.net
 * COPYRIGHT 2009-2015 SilverFoxServer.NET. All rights reserved.
 * MAIL:521266750@qq.com
 */
package net.silverfoxserver.core.socket;

import System.Xml.XmlDocument;
import java.net.InetAddress;
import java.net.InetSocketAddress;
import org.jboss.netty.channel.ChannelEvent;
import org.jboss.netty.channel.MessageEvent;

//

public class SessionMessage
{

	private ChannelEvent _e;
	private XmlDocument _doc;

	//private boolean _fromDBServer;
	private boolean _fromRCServer;

	/** 
	 
	 
	 @param strIpPort_ ip
	 @param doc_
	 @param fromServer_ 是否是同组服务器发来的消息，因现在是共用一个处理队列，防止客户端伪造
	*/
	public SessionMessage(ChannelEvent e_, XmlDocument doc_, boolean fromDBServer_, boolean fromRCServer_)
	{
		_e = e_;
		_doc = doc_;

		//
		if (true == fromDBServer_ && true == fromRCServer_)
		{
			throw new IllegalArgumentException("can not be all true!");
		}

		//_fromDBServer = fromDBServer_;
		_fromRCServer = fromRCServer_;
	}

	public String strIpPort()
	{
            
            InetSocketAddress remoteAddress = (InetSocketAddress)_e.getChannel().getRemoteAddress();
           
            String ippo = remoteAddress.getAddress().getHostAddress() + ":" + String.valueOf(remoteAddress.getPort());
            
            return ippo;
        
	}

	public XmlDocument doc()
	{
		return _doc;
	}
        
        public ChannelEvent e()
        {                
                return _e;
        }

	/** 
	 是否是同组服务器发来的消息，因现在是共用一个处理队列，防止客户端伪造
	 
	 @return 
	*/
	public  boolean getfromServer()
	{


	  return _fromRCServer;//true == _fromDBServer || true == 

	}


//	public  boolean getfromDBServer()
//	{
//		return _fromDBServer;
//	}

	public  boolean getfromRCServer()
	{
		return _fromRCServer;
	}

	public  String action()
	{
		return _doc.getDocumentElement().getChildren().get(0).getAttributeValue("action");
	}

        public void Dereference()
        {
        
            _e = null;
            _doc = null;
            
        }

}
